lib/kdi/kdi_helpers.rb

```diff
@@ -72,6 +72,31 @@ def create_kdi_asset(asset_hash, dup_check = true)
         asset_hash.compact
       end
 
+      def find_or_create_kdi_asset(asset_hash, match_key = nil)
+        uniq_asset_hash = uniq(asset_hash)
+        asset_hash_key = asset_hash.fetch(match_key)
+
+        # check to make sure it doesnt exist
+        a = if match_key.nil?
+              @assets.lazy.find { |asset| uniq(asset) == uniq_asset_hash }
+            else
+              @assets.lazy.find { |asset| asset[match_key] == asset_hash_key }
+            end
+
+        # SAnity check to make sure we are pushing data into the correct asset
+        unless a # && asset[:vulns].select{|v| v[:scanner_identifier] == args[:scanner_identifier] }.empty?
+          puts "Unable to find asset #{asset_hash}, creating a new one... "
+          create_kdi_asset(asset_hash)
```

Comment:
You can pass `true` here as the second argument to `create_kdi_asset` to skip the lookup in that method, since we already tried to look up the asset and know that it's missing.

Reply:
you meant pass false right? dup_check = false because we already know the asset exists? I added this boolean.

Reply:
:facepalm: True, I did mean `false` :stuck_out_tongue: